### DISCUSSION TITLE
pkg-config causes debug libraries to be linked in release builds

### DISCUSSION THREAD

**Discussion by dsvensson** [microsoft/vcpkg#37570] [Q&A]
For example:

`pkg_check_modules(PNG REQUIRED IMPORTED_TARGET libpng)`

With Visual Studio or Ninja Multi-Config, this leads to the debug version being linked. Not sure if it's because the pkg_check_modules doesn't set the `_RELEASE` and `_DEBUG` variants of the variables and thus overwrites each other, but the question remains. How should this be handled?

Bug in vcpkg? Bug in cmake? An annoying inconvenience with a workaround?

The `vcpkg_pkgconfig_get_modules` port seems to do something in this direction, but I haven't figured out if it can be used for regular builds or if it's just a ports helper.

**Comment by dg0yt** [comment]
Plain CMake `FindPkgConfig` doesn't support multi-config, basically because plain `pkg-config` doesn't support multi-config.

It would be possible to do per-config searches and to combine the result, maybe even in a convenient wrapper.

**Comment by moratom** [comment]
@dsvensson did you figure something out? It seems to me that a lot of ports suffer from linking to release versions of the library
